pyccel/parser/semantic.py

```diff
@@ -1407,21 +1436,23 @@ def format_shape(s):
         if var.precision == -1 and precision != var.precision:
             var.use_exact_precision()
 
-    def _assign_GeneratorComprehension(self, lhs_name, expr, **settings):
+    def _assign_GeneratorComprehension(self, lhs_name, expr):
         """
-        Visit the GeneratorComprehension node creating all necessary expressions
-        for its definition
+        Visit the GeneratorComprehension node.
+
+        Create all necessary expressions for the
+        GeneratorComprehension node definition.
 
         Parameters
         ----------
         lhs_name : str
-                    The name to which the expression is assigned
+                    The name to which the expression is assigned.
         expr : GeneratorComprehension
+                The GeneratorComprehension node.
 
-        Results
+        Returns
         -------
-        new_expr : CodeBlock
-                   CodeBlock containing the semantic version of the GeneratorComprehension node
+        pyccel.ast.variable.Variable
```

Comment:
Why did you remove the return description?

Reply:
Wrong input, fixed.

Reply:
I've just realised that this return type is wrong. See above

Reply:
It got fixed in [bce4623](https://github.com/pyccel/pyccel/pull/1385/commits/bce46231be73f3c7515d9765ec4808f5e557deda)